pkg/log/ansHook_test.go

```diff
@@ -0,0 +1,286 @@
+	hook, _ := newANSHook(ans.Configuration{}, "", &ansMock{})
+	assert.Equal(t, []logrus.Level{logrus.WarnLevel, logrus.ErrorLevel, logrus.PanicLevel, logrus.FatalLevel},
+		hook.Levels())
+}
+
+func TestANSHook_newANSHook(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		serviceKey    string
+		eventTemplate string
+	}
+	tests := []struct {
+		name                     string
+		args                     args
+		eventTemplateFileContent string
+		checkErr                 error
+		wantEvent                ans.Event
+		wantErrMsg               string
+	}{
+		{
+			name:      "Straight forward test",
+			args:      args{serviceKey: defaultServiceKeyJSON},
+			wantEvent: defaultEvent(),
+		},
+		{
+			name:       "No service key yields error",
+			wantErrMsg: "cannot initialize SAP Alert Notification Service due to faulty serviceKey json: error unmarshalling ANS serviceKey: unexpected end of JSON input",
+		},
+		{
+			name:       "Fails on check error",
+			args:       args{serviceKey: defaultServiceKeyJSON},
+			wantErrMsg: "check http request to SAP Alert Notification Service failed; not setting up the ANS hook: check failed",
+			checkErr:   fmt.Errorf("check failed"),
+		},
+		{
+			name:                     "With event template as file",
+			args:                     args{serviceKey: defaultServiceKeyJSON},
+			eventTemplateFileContent: `{"priority":123}`,
+			wantEvent:                mergeEvents(t, defaultEvent(), ans.Event{Priority: 123}),
+		},
+		{
+			name:      "With event template as string",
+			args:      args{serviceKey: defaultServiceKeyJSON, eventTemplate: `{"priority":123}`},
+			wantEvent: mergeEvents(t, defaultEvent(), ans.Event{Priority: 123}),
+		},
+		{
+			name:                     "With event template from two sources, string overwrites file",
+			args:                     args{serviceKey: defaultServiceKeyJSON, eventTemplate: `{"priority":789}`},
+			eventTemplateFileContent: `{"priority":123}`,
+			wantEvent:                mergeEvents(t, defaultEvent(), ans.Event{Priority: 789}),
+		},
+		{
+			name:       "Fails on validation error",
+			args:      args{serviceKey: defaultServiceKeyJSON, eventTemplate: `{"priority":-1}`},
+			wantErrMsg: "did not initialize SAP Alert Notification Service due to faulty event template json: Priority must be 1 or greater: event JSON failed the validation",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var testEventTemplateFilePath string
+			if len(tt.eventTemplateFileContent) > 0 {
+				testEventTemplateFilePath = writeTempFile(t, tt.eventTemplateFileContent)
+				defer os.Remove(testEventTemplateFilePath)
+			}
+
+			ansConfig := ans.Configuration{
+				ServiceKey:            tt.args.serviceKey,
+				EventTemplateFilePath: testEventTemplateFilePath,
+				EventTemplate:         tt.args.eventTemplate,
+			}
+			clientMock := &ansMock{checkErr: tt.checkErr}
+			defer clientMock.cleanup()
```

Comment:
is the cleanup neccessary?
   `clientMock := &ansMock{checkErr: tt.checkErr}`
creates a new struct and cleanup doen't free anything.

Reply:
true, done.